lib/daru/index.rb

```diff
@@ -386,5 +391,32 @@ def sparse_tuples
         [nil] * (cur.size - left.size) + left.map(&:first)
       }
     end
+
+    def tuples_with_rowspans
```

Comment:
In fact, I thought about just making it `# @private`-ly documented, it is not that easy to explain what it does and why does somebody need it :)